### PR TITLE
feat(daffio,design): highlight active tree item in sidebar

### LINKED ISSUE #2854: [FEAT] Add selected functionality for tree component
# :bulb: Feature request

## Feature Name
Add selected UI for tree component

## The Desired Behavior
A `[daff-tree-item]` that is active/selected should have a `selected` class added to it to properly style it.

## Your Use Case
As a designer, I would like to indicate to users if a tree item in a tree is currently selected/active.

## Environment

<pre><code>
Daffodil version: X.Y.Z
Angular version: X.Y.Z 

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: #2854 

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information
The way the options must be set is unfortunate. There is no way to set the options from the `    DaffTreeItemDirective` because of https://github.com/angular/angular/issues/50700#issuecomment-1590896233

### CODE REVIEW

**Comment by xelaint** [comment]
So the tree technically doesn't handle the "active" state unless `ROUTER_LINK_ACTIVE_CONFIG` is set in the implementation?

Is there a different way to do this?

**Comment by griest024** [comment]
> So the tree technically doesn't handle the "active" state unless `ROUTER_LINK_ACTIVE_CONFIG` is set in the implementation?
> 
> Is there a different way to do this?

no it does but the "overview" link will be active when a child guide is selected so you'll see 2 active items at once

**Comment by xelaint** [comment]
Also, getting this error in console in design-land:

```
template.html:1 ERROR Error: NG0311: Directive _RouterLinkActive does not have an input with a public name of daffTreeItem.
    at validateMappings (
```

**Comment by griest024** [comment]
> Also, getting this error in console in design-land:
> 
> ```
> template.html:1 ERROR Error: NG0311: Directive _RouterLinkActive does not have an input with a public name of daffTreeItem.
>     at validateMappings (
> ```

I cannot repro that

**Comment by xelaint** [comment]
> > So the tree technically doesn't handle the "active" state unless `ROUTER_LINK_ACTIVE_CONFIG` is set in the implementation?
> > Is there a different way to do this?
> 
> no it does but the "overview" link will be active when a child guide is selected so you'll see 2 active items at once

Why are the links below overview treated as child guides of overview? Shouldn't they be sibling to overview?

**Comment by griest024** [comment]
@xelaint I'll use cart as an example. the cart driver guide is at `/docs/packages/cart/drivers`. the cart "overview" guide has the link `/docs/packages/cart`. The angular router considers `/docs/packages/cart/drivers` to be a "child" of `/docs/packages/cart` and therefore considers `/docs/packages/cart` to be active on the route `/docs/packages/cart/drivers` *unless* `exact: true` is set.

File: libs/design/tree/src/tree-item/tree-item.directive.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import {
   Inject,
   Input,
 } from '@angular/core';
+import { RouterLinkActive } from '@angular/router';
```
Comment (damienwebdev) [inline libs/design/tree/src/tree-item/tree-item.directive.ts]: This makes the `design` library dependent on `@angular/router` which I've intentionally disallowed historically.

File: apps/daffio/src/app/docs/components/docs-list/docs-list.component.html [hunk 2]
```diff
@@ -5,7 +5,7 @@
     </ng-template>
 
     <ng-template #daffTreeItemTpl let-node>
-      <a daffTreeItem [node]="node" [routerLink]="node.url">{{ node.title }}</a>
+      <a [daffTreeItem]="ROUTER_LINK_ACTIVE_CONFIG" [node]="node" [routerLink]="node.url">{{ node.title }}</a>
```
Comment (damienwebdev) [inline apps/daffio/src/app/docs/components/docs-list/docs-list.component.html]: From a review of the directive's docs:

> <a routerLink="/user/bob" routerLinkActive #rla="routerLinkActive">
>   Bob {{ rla.isActive ? '(already open)' : ''}}
> </a>

Can you do:

> <a [selected]="rla.isActive"  #rla="routerLinkActive" [node]="node" [routerLink]="node.url">{{ node.title }}</a>

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/design/tree/src/tree-item/tree-item.directive.ts (hunk 1 above)
Comment (griest024) [reply libs/design/tree/src/tree-item/tree-item.directive.ts]: Then this feature cannot be implemented as requested. The best that we can do is set `daffTreeItem routerLinkActive="selected"` in app code.

File: apps/daffio/src/app/docs/components/docs-list/docs-list.component.html (hunk 2 above)
Comment (griest024) [reply apps/daffio/src/app/docs/components/docs-list/docs-list.component.html]: yes I did something like that

File: libs/design/tree/src/tree-item/tree-item.directive.ts (hunk 1 above)
Comment (griest024) [reply libs/design/tree/src/tree-item/tree-item.directive.ts]: removed everything from design

**Review by damienwebdev — APPROVED** [review]
lgtm

**Comment by xelaint** [comment]
I'd expect a parent tree item to be open if there is an active/selected tree item within it. Should this be done in this PR or create a separate issue for it?

**Comment by griest024** [comment]
> I'd expect a parent tree item to be open if there is an active/selected tree item within it. Should this be done in this PR or create a separate issue for it?

separate PR

**Review by xelaint — APPROVED** [review]

**Pull request merged**